k8s/client.go

```diff
@@ -242,16 +259,47 @@ func (c *client) createNamespaceSource() {
 }
 
 func (c *client) UpdateIngressStatus(ingress *v1beta1.Ingress) error {
-	ingressClient := c.clientset.ExtensionsV1beta1().Ingresses(ingress.Namespace)
+	ingressClient := c.ingressGetter.Ingresses(ingress.Namespace)
 
 	currentIng, err := ingressClient.Get(ingress.Name, metav1.GetOptions{})
 	if err != nil {
 		return err
 	}
 
 	currentIng.Status.LoadBalancer.Ingress = ingress.Status.LoadBalancer.Ingress
+	return c.updateIngressAndHandleConflicts(ingressClient, currentIng)
+}
+
+func (c *client) updateIngressAndHandleConflicts(ingressClient clientV1Beta1.IngressInterface, ingress *v1beta1.Ingress) error {
+	_, err := ingressClient.UpdateStatus(ingress)
+
+	switch {
+	case k8errors.IsConflict(err):
+		updatedIng, getErr := ingressClient.Get(ingress.Name, metav1.GetOptions{})
+		if getErr == nil && ingressStatusEqual(updatedIng.Status.LoadBalancer.Ingress, ingress.Status.LoadBalancer.Ingress) {
+			conflictingIngressStatusUpdates.Inc()
+			return nil
+		}
+		return err
```

Comment:
Isn't this sort of overkill? Conflict will happen if multiple feeds try to update the status, which we know will always happen at startup. Given it is expected, couldn't we just ignore the error if it is a conflict error? So we wouldn't need more than
```go
_, err := ingressClient.UpdateStatus(currentIng)
if !k8errors.IsConflict(err) {
  // ignore conflicts, as they are expected
  return err
}
return nil
```

Reply:
That's assuming the conflict is for the reason we expect. Unlikely though it may be that we'll see a conflict for another reason, I don't think the additional logic adds that much in the way of complexity.

Reply:
Okay, fair enough - would you mind adding a comment describing what it's checking? As I don't think it's super obvious why it returns nil in this case.